pkg/network/master/subnets.go

```diff
@@ -76,26 +80,29 @@ func (master *OsdnMaster) addNode(nodeName string, nodeIP string, hsAnnotations
 		}
 	}
 
-	// Create new subnet
-	sn, err := master.subnetAllocator.GetNetwork()
-	if err != nil {
-		return "", fmt.Errorf("error allocating network for node %s: %v", nodeName, err)
-	}
-
-	sub = &networkapi.HostSubnet{
-		TypeMeta:   metav1.TypeMeta{Kind: "HostSubnet"},
-		ObjectMeta: metav1.ObjectMeta{Name: nodeName, Annotations: hsAnnotations},
-		Host:       nodeName,
-		HostIP:     nodeIP,
-		Subnet:     sn.String(),
-	}
-	sub, err = master.osClient.HostSubnets().Create(sub)
-	if err != nil {
-		master.subnetAllocator.ReleaseNetwork(sn)
-		return "", fmt.Errorf("error creating subnet %s for node %s: %v", sn.String(), nodeName, err)
+	// Create new subet
+	for _, possibleSubnet := range master.subnetAllocatorList {
+		sn, err := possibleSubnet.GetNetwork()
+		if err != nil && err != netutils.ErrSubnetAllocatorFull {
+			return "", fmt.Errorf("error allocating network for node %s: %v", nodeName, err)
+		} else {
```

Comment:
When err != nil and err == netutils.ErrSubnetAllocatorFull, then we will use invalid `sn` here. 
We should do something like:
```
sn, err := possibleSubnet.GetNetwork()
if err ==  netutils.ErrSubnetAllocatorFull {
    // Current subnet exhausted, check next one
   continue
} else if err != nil {
  glog.Errorf("Error allocating network from subnet: %v", posssibleSubnet)
  continue
} else {
  // Process valid subnet
}
```